map-comparison/comparisons.py

```diff
@@ -0,0 +1,60 @@
+import os
+import json
+import psutil
+
+import click
+
+for k in ['OMP_NUM_THREADS', 'OPENBLAS_NUM_THREADS', 'MKL_NUM_THREADS']:
+    os.environ[k] = '1'
+
+import numpy as np
+# import dask.distributed as dd
+
+from NumpyBenchmark import NumpyBenchmark
+from LiberTEMBenchmark import LiberTEMBenchmark
+from DaskBenchmark import DaskBenchmark
+
+def _preload():
+    import libertem.preload  # NOQA
+
+@click.command()
+@click.option('--path', 
+    default=r'testfile.raw')
+@click.option('--scheduler-uri', default=None)
+@click.option('--stackheight', default=32)
+@click.option('--scan-size', default="64,64")
+@click.option('--detector-size', default="128,128")
+@click.option('--dtype', default="float32")
+@click.option('--method', type=click.Choice(['direct', 'mmap', 'read']), required=True)
+@click.option('--num-workers', default=psutil.cpu_count(logical=False))
+@click.option('--num-nodes', default=None, type=int)
+@click.option('--warmup-rounds', default=1)
+def main(path, scheduler_uri, stackheight, dtype, scan_size, detector_size, method, num_workers, num_nodes,
+         warmup_rounds):
+    scan_size = tuple(int(x) for x in scan_size.split(","))
+    detector_size = tuple(int(x) for x in detector_size.split(","))
+    (a, b) = scan_size
+    (f, g) = detector_size
+    
+    roi = np.zeros(shape=scan_size, dtype=np.bool)
+    roi[(a * 5) // 10:(a * 6) // 10, (b * 2) // 10:(b * 3) // 10] = True
+    
+    mask = np.zeros(shape=detector_size, dtype=np.float32)
+    mask[(f * 5) // 10:(f * 6) // 10, (g * 2) // 10:(g * 3) // 10] = 1
+
+    numpy_bench = NumpyBenchmark(path, dtype, scan_size, detector_size, warmup_rounds, roi, mask)  
+    print(json.dumps(numpy_bench.bench_all()))
+
+    libertem_bench = LiberTEMBenchmark(path, dtype, scan_size, detector_size, warmup_rounds, roi, mask)
```

Comment:
Is the dask local cluster ever teared down/scaled to zero workers?

Reply:
Aaah, dask was automatically using the LiberTEM dask cluster!

http://distributed.dask.org/en/latest/client.html
> The Client registers itself as the default Dask scheduler, and so runs all dask collections like dask.array, dask.bag, dask.dataframe and dask.delayed

I'd prefer an explicit call or reference for that, if it was me. With that it is very difficult to control which scheduler is actually executing the commands. It may very easily happen that someone sends a couple of TB or sensitive data across the globe by accident, simply because somewhere in the application someone created a Dask.distributed Client that connects to a remote cluster, which then hijacks all subsequent commands. 👀